### PR TITLE
[10034] Update the release process documentation.

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope

This is the proposed release process once we autimated the downloads and docs with GitHub Actions + PyPI  + Read the docs.

Here you can see the result https://twisted--1465.org.readthedocs.build/en/1465/core/development/policy/release-process.html

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10034
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I ran additional checks listed at [Getting Your Patch Accepted](https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [ ] I have updated the automated tests.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

File: .readthedocs.yml
Comment (graingert) [inline .readthedocs.yml]: I think it's worth enabling this once we can ignore the warnings in ToC https://github.com/sphinx-doc/sphinx/pull/8289

(Sphinx 3.3.0)

File: docs/core/development/policy/release-process.rst
Comment (graingert) [inline docs/core/development/policy/release-process.rst:204]: I think it's worth publishing that the releases are no longer signed

**Review by graingert — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE
Comment (graingert) [inline .github/PULL_REQUEST_TEMPLATE:15]:
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE
Comment (graingert) [inline .github/PULL_REQUEST_TEMPLATE:19]:
```suggestion
```

I think it's worth removing this and just relying on CI here

Comment (mthuurne) [reply .github/PULL_REQUEST_TEMPLATE:19]: I do think it's useful to mention that command and link the coding standard. Maybe it doesn't have to be part of the checklist though.

The same applies for "additional checks" and "created a newsfragment": these are checked by CI, but it's good to have the links to the documentation immediately available.

Comment (exarkun) [reply .github/PULL_REQUEST_TEMPLATE:19]: What does this change have to do with the release process at all?

File: .github/PULL_REQUEST_TEMPLATE
Comment (adiroiban) [reply .github/PULL_REQUEST_TEMPLATE:15]: This was intentional. The reasoning is.

2 empty lines before a section title. 1 empty line after a section title 

In this way, I find it easier to detect sections :)

File: .github/PULL_REQUEST_TEMPLATE
Comment (adiroiban) [reply .github/PULL_REQUEST_TEMPLATE:19]: Well... if this change would have not been controversial, I would be happy to have it as drive by fix.

I think that now is better to do `pre-commit install`.

@graingert we can have a separate PR in which we try to improve the Contributing end to end :)

File: .readthedocs.yml
Comment (adiroiban) [reply .readthedocs.yml]: Right now, it is blocked by a different issue.

When fail on warnings is True, read the docs will not build the HTML files..so you can't check the partial output.

I was expecting that this PR will fix it https://github.com/readthedocs/readthedocs.org/pull/7251 

but when sphinx build fails, I still don't get HTML files

File: docs/core/development/policy/release-process.rst
Comment (adiroiban) [reply docs/core/development/policy/release-process.rst:204]: Yes. I think that this should be top level info for the next release together with the fact that 2.7 is no longer supported.

**Comment by adiroiban** [comment]
As there are no major complains, I am going to merge this as soon as all the tests are green.
I plan to practice what is written here and maybe do a 21.1.0 release.

**Comment by rodrigc** [comment]
@adiroiban It was not good practice for you to submit this change to the release process documentation, and merge it after @graingert approved very quickly.

You are not the current release engineer for Twisted, and @graingert does not have a track record as a release engineer for the Twisted project.

You submitted this change right in the middle of a time when a lot of people are on holiday.
What you did was procedurally correct by the Twisted development process,
but it is not very nice.

**Comment by adiroiban** [comment]
Feel free to revert it.

I have requested the review on 23 of Oct and merged, it was approved by graingert on 3rd of Nov and I merged on the 24 of Dec.

Good luck

**Pull request merged**
